### DISCUSSION TITLE
Utilizing a compiler plugin for code generation vs SBT source generators

### DISCUSSION THREAD

**Discussion by kincsescsaba** [sbt/sbt#7646] [Q&A]
Hey there, I'd be happy to know what you would recommend considering the below details:

I need to create a dynamic DSL generator that generates a DSL based on the existing codebase. This looks like a multi-phase process that needs to query the whole codebase, so a compiler plugin seems like a good fit. However, I wonder if what a compiler plugin would generate would be visible for the IDE. Regarding SBT source generators, I know at least how they work and can be treated as supported by IntelliJ IDE, as the IDE recognizes the unmanaged source directory.
I would be open to a compiler plugin-based solution that utilizes the SBT setting of an unmanaged source directory.

So, beyond my general question, I'd be happy to read your findings on whether using a compiler plugin to target the SBT-configured unmanaged source directory with generated sources would be a good idea.

**Comment by eed3si9n** [comment]
It might depend on the querying process as well. If you need compile-time information from the compiler or affect the way things are compiled you probably need a compiler plugin or a macro, otherwise a code generator might be simpler.
